src/fsharp/vs/Exprs.fsi

```diff
@@ -206,4 +206,7 @@ module BasicPatterns =
     /// Matches expressions for an unresolved call to a trait 
     val (|TraitCall|_|) : FSharpExpr -> (FSharpType list * string * FSharpType list * FSharpType list * FSharpExpr list) option 
 
+module DerivedPatterns =
 
+    /// Matches expressions for an unresolved call to a trait with MemberFlags info
+    val (|TraitCallExtended|_|) : FSharpExpr -> (FSharpType list * string * Ast.MemberFlags * FSharpType list * FSharpType list * FSharpExpr list) option
```

Comment:
You can just change TraitCall's signature.  We will bump the major version number